### DISCUSSION TITLE
Multiple components with 2 phase flow?

### DISCUSSION THREAD

**Discussion by JonathanEdelmanS** [idaholab/moose#28370] [Q&A Modules: Porous Flow]
What would be the best way to model a pipe being used to pump liquid CO2 into a porous rock with brine? We want simple fluid dynamics in the pipe, but it seems the THM module's components only has support for one phase flow at the moment.

**Comment by GiudGiud** [comment]
Hello

There are some examples in the porous_flow module for this problem.
THM is mainly designed for nuclear engineering flow loops at this point. 
EDIT: this is arguable. Let's say that it is designed for single phase compressible fluids. 

Guillaume

**Reply by JonathanEdelmanS** [reply]
Hello Guillaume

Which examples consider the fluid flow in the pipe as well as the rock?

Jonathan

**Reply by GiudGiud** [reply]
@cpgr are there any?
